instrumentation/graphql-java-12.0/library/src/main/java/io/opentelemetry/instrumentation/graphql/OpenTelemetryInstrumentation.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.graphql;
+
+import graphql.ExecutionResult;
+import graphql.GraphQLError;
+import graphql.execution.instrumentation.InstrumentationContext;
+import graphql.execution.instrumentation.InstrumentationState;
+import graphql.execution.instrumentation.SimpleInstrumentation;
+import graphql.execution.instrumentation.SimpleInstrumentationContext;
+import graphql.execution.instrumentation.parameters.InstrumentationExecuteOperationParameters;
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters;
+import graphql.language.AstPrinter;
+import graphql.language.AstTransformer;
+import graphql.language.BooleanValue;
+import graphql.language.EnumValue;
+import graphql.language.Node;
+import graphql.language.NodeVisitorStub;
+import graphql.language.NullValue;
+import graphql.language.OperationDefinition;
+import graphql.language.OperationDefinition.Operation;
+import graphql.language.Value;
+import graphql.language.VariableReference;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+import graphql.util.TreeTransformerUtil;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+
+public final class OpenTelemetryInstrumentation extends SimpleInstrumentation {
```

Comment:
Can be package private
```suggestion
final class OpenTelemetryInstrumentation extends SimpleInstrumentation {
```

Reply:
it is used from `GraphqlSingletons` to figure out whether it is already present in instrumentation chain

Reply:
Hmm, right. Kinda nit: we could use `getClass()` to compare the instrumentations instead of making this public.